test/base_maths.jl

```diff
@@ -35,6 +35,10 @@ using Test
             @test b1 + Matrix(b1) == b1 + b1
             @test_throws DimensionMismatch b1 + Matrix(b3)
 
+            # Test on non-square blocks
+            # https://github.com/invenia/BlockDiagonals.jl/issues/124
+            @test bns + Matrix(bns) == Matrix(bns) + Matrix(bns)
```

Comment:
Could we add a test where blocks are different sizes as well?

Reply:
Do you mean something like

```Julia
"BlockDiagonal + BlockDiagonal"
@test b1 + b2 == Matrix(b1) + Matrix(b2)
```

Not all the blocks of `bns` are the same size, and for `BlockDiagonal + StridedMatrix` only one has blocks.

Reply:
I mean something like

```
b = BlockDiagonal([rand(2, 3), rand(3, 2)])
m = rand(5, 5)

```

Reply:
Yep, that makes sense to add